### ISSUE TITLE
Step with "Continue on Error" does not end if step fails

### ISSUE THREAD

**Issue by kanoutaisuke** [Microsoft/vso-agent#311]
Hi, i found 2 issues around this same problem, and the other 2 seem to be slightly different.
In my build, I have continue on error checked for my run tests step, yet it doesnt continue. It just continuously runs that same step over and over, never moving on to my publish test results step. Anyone else come across this issue before?

**Comment by bryanmacfarlane** [comment]
This is fixed in the replacement agent.  This agent never got that feature: https://github.com/Microsoft/vsts-agent

**Comment by kanoutaisuke** [comment]
Let me rephrase this a little, It runs the step over and over until it hits the 1 hour timeout and then fails the entire build. I can remote in and see it running the tests, but it hangs up on most and fails them all. It consistently says 0 passed in the log, even though I viewed 1 that passed for sure. On top of that, even if it hits the 1 hour limit currently set, should the publish results step still go off since it's set to always run?

**Comment by ericsciple** [comment]
@kanoutaisuke This agent is deprecated. Can you try the new agent: https://github.com/Microsoft/vsts-agent

**Comment by kanoutaisuke** [comment]
I understand this one is deprecated, accidentally posted this in this one as this was where I saw the similar questions. I'm not using the deprecated agent to begin with.
